src/interface/variable_pack.hpp

```diff
@@ -233,6 +256,40 @@ void FillVarView(const vpack_types::VarList<T> &vars, PackIndexMap *vmap,
   Kokkos::deep_copy(vector_component, host_vc);
 }
 
+template <typename T>
+void FillSwarmVarView(const vpack_types::SwarmVarList<T> &vars, PackIndexMap *vmap,
+                      ViewOfParArrays<T> &cv) {
+  using vpack_types::IndexPair;
+
+  auto host_view = Kokkos::create_mirror_view(Kokkos::HostSpace(), cv);
+
+  int vindex = 0;
+  int sparse_start;
+  std::string sparse_name;
+  // TODO(BRR) Remove the logic for sparse variables
+  for (const auto v : vars) {
+    if (vmap != nullptr) {
+      vmap->insert(std::pair<std::string, IndexPair>(
+          sparse_name, IndexPair(sparse_start, vindex - 1)));
+      sparse_name = "";
+    }
+    int vstart = vindex;
+    for (int k = 0; k < v->GetDim(6); k++) {
+      for (int j = 0; j < v->GetDim(5); j++) {
+        for (int i = 0; i < v->GetDim(4); i++) {
+          host_view(vindex++) = v->data.Get(k, j, i);
+        }
+      }
+    }
```

Comment:
Note to self: why is this a 3D loop when the `ParticleVariable`s are all 1D? Maybe this is fine because these `GetDim`s are all 1?